components/dashboard/src/teams/JoinTeam.tsx

```diff
@@ -23,20 +23,7 @@ export default function () {
                     throw new Error("This invite URL is incorrect.");
                 }
 
-                let team;
-                try {
-                    team = await getGitpodService().server.joinTeam(inviteId);
-                } catch (error) {
-                    const message: string | undefined = error && typeof error.message === "string" && error.message;
-                    const regExp = /You are already a member of this team. \((.*)\)/;
-                    const match = message && regExp.exec(message);
-                    if (match && match[1]) {
-                        const slug = match[1];
-                        history.push(`/t/${slug}/members`);
-                        return;
-                    }
-                    throw error;
-                }
+                const team = await getGitpodService().server.joinTeam(inviteId);
```

Comment:
By removing the `try` to do we risk ignoring other errors besides the re-joining error?

Reply:
Only the "already a member" is handled and the rest are then re-thrown and caught by the outer try, which we're keeping around. Here, the behaviour should remain the same (minus the explicit error string compare)